3-extra/1-guess-the-number-project/script.js

```diff
@@ -1,35 +1,77 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable no-plusplus */
+/* eslint-disable max-len */
 let randomNumber = Math.floor(Math.random() * 100 + 1);
+let guessCount = 7;
+let isWon = false;
+const numberInput = document.querySelector('.inputs-Values');
+const finalOutput = document.querySelector('.final-output');
+const triesOutput = document.querySelector('.tries-output');
+document.querySelector('.btnNewGame').addEventListener('click', newGame);
 
 function guessNumber() {
-  //Collect input from the user
-  let guess = document.querySelector(".inputs-Values").value;
+  // Collect input from the user
+  const guess = parseInt(numberInput.value, 10);
 
-  //If the user inputs a bad input ie 0, empty string, number greater that 100, number less than zero Print "Please enter a number between 1 and 100"
+  if (typeof guess !== 'number' || guess <= 0 || guess > 100) {
+    finalOutput.innerText = 'Please enter a number between 1 and 100';
+    return;
+  }
+  if (guess !== randomNumber && guessCount <= 1) {
+    triesOutput.innerText = '';
+    finalOutput.innerText = `You Lose, the number was ${randomNumber}`;
+    return;
+  }
+  if (guess > randomNumber) {
+    guessCount--;
+    triesOutput.innerText = `Number of Tries: ${guessCount}`;
+    finalOutput.innerText = 'Number is too high, try again';
+    return;
+  }
+  if (guess < randomNumber) {
+    guessCount--;
+    triesOutput.innerText = `Number of Tries: ${guessCount}`;
+    finalOutput.innerText = 'Number is too low, try again';
+    return;
+  }
+  if (guess === randomNumber) {
+    guessCount -= !isWon ? 1 : 0;
+    triesOutput.innerText = `It took you ${7 - guessCount} tries`;
+    finalOutput.innerText = `${guess} is correct. You win!`;
+    isWon = true;
+  }
```

Comment:
To avoid mixing text and code, you can refactor this code to take text strings from an enumerator or from a set of variables at the beginning of the program.

More advanced version of the code: You can also refactor the error conditions into their own functions array and then check all of them using an array function. Something similar to this:
```
const isInvalid = Object.values(invalidInputConditions).some((checkInputIsInvalid) =>
    checkInputIsInvalid(guess)
);
```